### PR TITLE
[kicker.de] Add extractor

### LINKED ISSUE #3670: [kicker.de] site support request
### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.04.08** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

Germany

### Example URLs

- Single Video: https://www.kicker.de/rose-erklaert-das-bringt-schlotterbeck-dem-bvb-901149/video

### Description

Site www.kicker.de has some short interview videos (kicker.tv), which seem to be delivered by dailymotion. All are playable in a browser but fail to download with yt-dlp. Maybe georestricted (<script src="https://geo.dailymotion.com/libs/player/xslz.js"></script>). No idea how to check, if there is DRM involved...

### Verbose log

```shell
$ yt-dlp --ignore-config --verbose https://www.kicker.de/rose-erklaert-das-bringt-schlotterbeck-dem-bvb-901149/video
[debug] Command-line config: ['--ignore-config', '--verbose', 'https://www.kicker.de/rose-erklaert-das-bringt-schlotterbeck-dem-bvb-901149/video']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.04.08 [7884ade65]
[debug] Python version 3.9.10 (CPython 64bit) - macOS-10.13.6-x86_64-i386-64bit
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 4.4 (setts), ffprobe 4.4
[debug] Optional libraries: brotli, certifi, Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [generic] Extracting URL: https://www.kicker.de/rose-erklaert-das-bringt-schlotterbeck-dem-bvb-901149/video
[generic] video: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] video: Downloading webpage
[generic] video: Extracting information
[debug] Looking for video embeds
ERROR: Unsupported URL: https://www.kicker.de/rose-erklaert-das-bringt-schlotterbeck-dem-bvb-901149/video
Traceback (most recent call last):
  File "/Users/whoever/Library/Python/3.9/lib/python/site-packages/yt_dlp/YoutubeDL.py", line 1408, in wrapper
    return func(self, *args, **kwargs)
  File "/Users/whoever/Library/Python/3.9/lib/python/site-packages/yt_dlp/YoutubeDL.py", line 1478, in __extract_info
    ie_result = ie.extract(url)
  File "/Users/whoever/Library/Python/3.9/lib/python/site-packages/yt_dlp/extractor/common.py", line 641, in extract
    ie_result = self._real_extract(url)
  File "/Users/whoever/Library/Python/3.9/lib/python/site-packages/yt_dlp/extractor/generic.py", line 4039, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.kicker.de/rose-erklaert-das-bringt-schlotterbeck-dem-bvb-901149/video
```

### PR DESCRIPTION
**Pull request by HobbyistDev**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please **make sure the description explains the purpose and effect** of your *pull request* and is worded well enough to be understood. Provide as much **context and examples** as possible.

Region : Germany

Add extractor for kicker.de. The sites provide sports article and video. The video service itself called kicker.tv. In this PR, the extractor will extract the `dailymotion_video_id` from `dmprivateid` in the webpage and then pass the `dailymotion_video_id` to `dailymotion.py`

Close #3670

### CODE REVIEW

File: yt_dlp/extractor/kicker.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        dailymotion_video_id = self._search_regex(
+            r'data-dmprivateid=\"(?P<video_id>\w+)\"', webpage,
+            'video_id', group='video_id')
+
+        return {
+            '_type': 'url_transparent',
+            'ie': DailymotionIE.ie_key(),
+            'url': f'https://www.dailymotion.com/video/{dailymotion_video_id}',
+            'title': self._html_extract_title(webpage),
+        }
```
Comment (pukkandan) [inline yt_dlp/extractor/kicker.py:60]: self.url_result

File: yt_dlp/extractor/kicker.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            'upload_date': '20220519',
+            'uploader': 'kicker.de',
+            'description': 'md5:0c2060c899a91c8bf40f578f78c5846f',
+            'ie': 'Dailymotion',
+            'like_count': int,
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_slug = self._match_valid_url(url).group('video_slug')
```
Comment (pukkandan) [inline yt_dlp/extractor/kicker.py:48]: name the group `id` and use self._match_id(url)

File: yt_dlp/extractor/kicker.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            'duration': 331,
+            'timestamp': 1652966015,
+            'thumbnail': r're:https?://s\d+\.dmcdn\.net/v/TxU4Z1YYCmtisTbMq/x1080',
+            'tags': ['FELIX KROOS', 'EINFACH MAL LUPPEN', 'KROOS', 'FSV FORTUNA PANKOW', 'published', 'category.Amateurs', 'marketingpreset.Spreekick'],
+            'age_limit': 0,
+            'view_count': int,
+            'upload_date': '20220519',
+            'uploader': 'kicker.de',
+            'description': 'md5:0c2060c899a91c8bf40f578f78c5846f',
+            'ie': 'Dailymotion',
```
Comment (pukkandan) [inline yt_dlp/extractor/kicker.py:42]:
```suggestion
```
same above

File: yt_dlp/extractor/_extractors.py [hunk 4]
```diff
@@ -491,7 +491,6 @@
     FancodeVodIE,
     FancodeLiveIE
 )
-
```
Comment (pukkandan) [inline yt_dlp/extractor/_extractors.py:494]: revert. This will cause unnecessary conflicts

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/kicker.py (hunk 1 above)
Comment (HobbyistDev) [reply yt_dlp/extractor/kicker.py]: Fixed

File: yt_dlp/extractor/kicker.py (hunk 2 above)
Comment (HobbyistDev) [reply yt_dlp/extractor/kicker.py]: Fixed

File: yt_dlp/extractor/kicker.py (hunk 3 above)
Comment (HobbyistDev) [reply yt_dlp/extractor/kicker.py]: Fixed

File: yt_dlp/extractor/_extractors.py (hunk 4 above)
Comment (HobbyistDev) [reply yt_dlp/extractor/_extractors.py]: Fixed

File: yt_dlp/extractor/kicker.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    def _real_extract(self, url):
+        video_slug = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_slug)
+        dailymotion_video_id = self._search_regex(
+            r'data-dmprivateid=\"(?P<video_id>\w+)\"', webpage,
+            'video_id', group='video_id')
+
+        return self.url_result(
```
[the remark this answers is not in the document]
Comment (HobbyistDev) [reply yt_dlp/extractor/kicker.py:53]: I think the `url_transparent` is not required as `DailymotionIE` give more accurate result than the webpage result

File: yt_dlp/extractor/kicker.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            'like_count': int,
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_slug = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_slug)
+        dailymotion_video_id = self._search_regex(
+            r'data-dmprivateid=\"(?P<video_id>\w+)\"', webpage,
```
Comment (pukkandan) [inline yt_dlp/extractor/kicker.py:50]:
```suggestion
            r'data-dmprivateid\s*=\s*[\'"](?P<video_id>\w+)', webpage,
```

File: yt_dlp/extractor/kicker.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            'video_id', group='video_id')
```
Comment (pukkandan) [inline yt_dlp/extractor/kicker.py:51]:
```suggestion
            'video id', group='video_id')
```

**Review by pukkandan — APPROVED** [review]

**Pull request merged**